src/ray/raylet/scheduling/cluster_resource_data.h

```diff
@@ -39,6 +39,167 @@ std::vector<FixedPoint> VectorDoubleToVectorFixedPoint(const std::vector<double>
+  /// \param capacity: numeric capacity value for the resource to add.
+  /// \return True, if the resource was successfully added. False otherwise.
+  void AddOrUpdateResource(const std::string &resource_name, const FixedPoint &capacity);
+
+  /// \brief [deprecated] Delete a resource from the resource set. This method is departed,
+  /// kept just for back compatibility and will be removed in future. Please do not use
+  /// it.
+  ///
+  /// \param resource_name: name/label of the resource to delete.
+  /// \return True if the resource was found while deleting, false if the resource did not
+  /// exist in the set.
+  bool DeleteResource(const std::string &resource_name);
+
+  /// \brief Getter method for custom_resources_
+  const std::unordered_map<int64_t, FixedPoint> &GetCustomResourceAmountMap() const;
+
+  /// \brief Add a set of resources to the current set of resources subject to upper
+  /// limits on capacity from the total_resource set.
+  ///
+  /// \param other: The other resource set to add.
+  /// \param total_resources: Total resource set which sets upper limits on capacity for
+  /// each label.
+  void AddResourcesCapacityConstrained(const ResourceSet &other,
+                                       const ResourceSet &total_resources);
+
+  /// \brief Aggregate resources from the other set into this set, adding any missing
+  /// resource labels to this set.
+  ///
+  /// \param other: The other resource set to add.
+  /// \return Void.
+  void AddResources(const ResourceSet &other);
+
+  /// \brief Subtract a set of resources from the current set of resources and
+  /// check that the post-subtraction result nonnegative. Assumes other
+  /// is a subset of the ResourceSet. Deletes any resource if the capacity after
+  /// subtraction is zero.
+  ///
+  /// \param other: The resource set to subtract from the current resource set.
+  /// \return Void.
+  void SubtractResources(const ResourceSet &other);
+
+  /// \brief Same as SubtractResources but throws an error if the resource value
+  /// goes below zero.
+  ///
+  /// \param other: The resource set to subtract from the current resource set.
+  /// \return Void.
+  void SubtractResourcesStrict(const ResourceSet &other);
+
+  /// \brief [deprecated] Return the capacity value associated with the specified
+  /// resource.This method is departed, kept just for back compatibility and will be
+  /// removed in future. Please do not use it.
+  ///
+  /// \param resource_name: Resource name for which capacity is requested.
+  /// \return The capacity value associated with the specified resource, zero if resource
+  /// does not exist.
+  FixedPoint GetResource(const std::string &resource_name) const;
+
+  /// \brief Set predefined resource quantity by index.
+  void SetPredefinedResources(PredefinedResources resource, FixedPoint quantity);
+
+  /// Return the number of CPUs.
+  ///
+  /// \return Number of CPUs.
+  const ResourceSet GetNumCpus() const;
```

Comment:
nits: return const value [probably doesn't make a lot of sense](https://stackoverflow.com/questions/6299967/what-are-the-use-cases-for-having-a-function-return-by-const-value-for-non-built) in C++ 11+ era.